### PR TITLE
Fix txn near cache

### PR DESCRIPTION
**Pull request by ahmetmircik**
Fixes: https://github.com/hazelcast/hazelcast/issues/16577
Backport of: https://github.com/hazelcast/hazelcast/pull/16579
ee: https://github.com/hazelcast/hazelcast-enterprise/pull/3514

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/map/impl/tx/TransactionalMapProxy.java [hunk 1]
```diff
@@ -143,19 +145,19 @@ public Object put(Object key, Object value, long ttl, TimeUnit timeUnit) {
...
-        try {
-            Data keyData = mapServiceContext.toData(nearCacheKey, partitionStrategy);
-            Object valueBeforeTxn = toObjectIfNeeded(putInternal(keyData, mapServiceContext.toData(value), ttl, timeUnit));
-
-            TxnValueWrapper currentValue = txMap.get(keyData);
-            Type type = valueBeforeTxn == null ? Type.NEW : Type.UPDATED;
-            TxnValueWrapper wrapper = new TxnValueWrapper(value, type);
-            txMap.put(keyData, wrapper);
-            return currentValue == null ? valueBeforeTxn : checkIfRemoved(currentValue);
-        } finally {
```
Comment (petrpleshachkov) [inline hazelcast/src/main/java/com/hazelcast/map/impl/tx/TransactionalMapProxy.java]: If some exceptions happen in this block, we still call invalidation in the `finally`. We don't do this in the new code. Is it OK if we have an exceptional scenario? It seems in the new implementation we handle failures only on commit.

Comment (ahmetmircik) [reply hazelcast/src/main/java/com/hazelcast/map/impl/tx/TransactionalMapProxy.java]: Yes it is ok not to have that finally block. The right place to apply invalidations is commit phase. Depending on the failure or success near cache interaction should happen in there.

**Comment by mmedenjak** [comment]
run-lab-run

**Comment by ahmetmircik** [comment]
Merging this PR, all tests were passed successfully, PR builder has some other issues.

**Pull request merged**